components/dashboard/src/user-settings/Preferences.tsx

```diff
@@ -8,14 +8,42 @@ import { useCallback, useContext, useState } from "react";
 import { getGitpodService } from "../service/service";
 import { UserContext } from "../user-context";
 import { trackEvent } from "../Analytics";
-import SelectIDE from "./SelectIDE";
 import { PageWithSettingsSubMenu } from "./PageWithSettingsSubMenu";
 import { ThemeSelector } from "../components/ThemeSelector";
 import Alert from "../components/Alert";
 import { Link } from "react-router-dom";
 import { Heading2, Subheading } from "../components/typography/headings";
 import { useUserMaySetTimeout } from "../data/current-user/may-set-timeout-query";
 import { Button } from "../components/Button";
+import { User } from "@gitpod/gitpod-protocol";
+import SelectIDE from "./SelectIDE";
+
+export type IDEChangedTrackLocation = "workspace_list" | "workspace_start" | "preferences";
+
+export const updateUserIDEInfo = async (
```

Comment:
This looks like it's not used anywhere. Was it copied over from `SelectIDE.tsx` during dev and then forgot to remove maybe?

Not something we need to do here, but figured I'd comment for context: FWIW for any new user-related updates, I'd love to start moving calls to use react-query mutations. We've got one that we setup w/ the new onboarding flow that can be used the same way. ATM we still have to update UserContext, but we can move away from that as we swap more things to react-query for the user. https://github.com/gitpod-io/gitpod/blob/3f8e507cb7dccc0b259f3152669390bdd41dd7e7/components/dashboard/src/onboarding/UserOnboarding.tsx#L38.